### DISCUSSION TITLE
[Q] Why is it `0`?

### DISCUSSION THREAD

**Discussion by anzhi0708** [ocaml/ocaml#12113] [Q&A]
Hello, here are some n00b code for u:

```ocaml
let money = ref 0 in

let t1 = Domain.spawn (
  fun _ ->
    for i = 1 to 100 do
      money := !money + 1
    done
)
in

let t2 = Domain.spawn (
  fun _ ->
    for i = 1 to 100 do
      money := !money - 1
    done
) in

Domain.join t1; Domain.join t2;

print_int !money;
print_endline "";


```

I was expecting var `money` to equal to something else than 0?

**Comment by kayceesrk** [comment]
Since the question isn't directly related to the OCaml compiler development, may I suggest asking this question on the more appropriate [discuss.ocaml.org](https://discuss.ocaml.org/) forum. 

Closing this.
